w3lib/url.py

```diff
@@ -204,8 +204,15 @@ def _add_or_replace_parameters(url, params):
     parsed = urlsplit(url)
     args = parse_qsl(parsed.query, keep_blank_values=True)
 
-    new_args = OrderedDict(args)
-    new_args.update(params)
+    new_args = []
+    changing_params = list(params.keys())
```

Comment:
why is it converted to a list? a set would be more correct and `.keys()` method call is not required: `changing_params = set(params)`.